result/bgpv4prefix.yaml

```diff
@@ -0,0 +1,55 @@
+components:
+  schemas:
+    Bgpv4Prefix.State.Request:
+      description: >-
+        The request to retrieve IPv4 Prefix state of a BGPV4 Peer.
+      type: object
+      properties:
+        ethernet_names:
+          description: >-
+            The names of BGPV4 Peers for which IPV4 Prefix state will be retrieved.
+            If no names are specified then the results will contain IPV4 Prefix
+            state for all configured BGPV4 Peers.
+          type: array
+          items:
+            type: string
+          x-constraint:
+            - "/components/schemas/Bgp.V4Peer/properties/name"
+
+    Bgpv4Prefix.State:
+      description: >-
+        BGP IPv4 Prefix state.
+      type: object
+      properties:
+        bgp_peer_name:
+          description: >-
+            The name of the BGPV4 Peer associated with the prefix state information.
+          type: string
+        prefixes:
+          type: array
+          items:
+            $ref: '#/components/schemas/Bgpv4Prefix.Prefix'
+
+    Bgpv4Prefix.Prefix:
```

Comment:
In OTG Bgp.V4RouteRange, today we already support 'Path_id', 'AS_Path', 'Community', etc. If first two can be also be supported in this?

Reply:
Something in my mind and how it will look if a test client just will need to request through the OTG API “Get BGP IPv4 Unicast Routes”, “Get BGP IPv6 Unicast Routes”, “Get ISIS IPv4 Prefix Routes” or “Get ISIS IPv6 Prefix Routes”.
In response it will get “properties” of string key-values pairs array.
One such row entry of the array can be “{key-1:value-1, key-2:value-2, …,key-n:value-n}”.
For an example, for “Get BGP IPv4 Unicast Routes”:
“[IPv4 Prefix:202.1.1.0, IPv4 Next Hop:200:0:0:1, Path ID:10, AS Path:{1}]”

Then we don’t care about numbers fields to be added in future.

Reply:
Something in my mind and how it will look if a test client just will need to request through the OTG API “Get BGP IPv4 Unicast Routes”, “Get BGP IPv6 Unicast Routes”, “Get ISIS IPv4 Prefix Routes” or “Get ISIS IPv6 Prefix Routes”.
In response it will get “properties” of string key-values pairs array.
One such row entry of the array can be “{key-1:value-1, key-2:value-2, …,key-n:value-n}”.
For an example, for “Get BGP IPv4 Unicast Routes”:
“[IPv4 Prefix:202.1.1.0, IPv4 Next Hop:200:0:0:1, Path ID:10, AS Path:{1}]”

Then we don’t care about numbers fields to be added in future.